src/libraries/System.IO.FileSystem/tests/RandomAccess/FlushToDisk.Unix.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class RandomAccess_FlushToDisk : RandomAccess_Base<long>
+    {
+        [Fact]
+        public void FlushFileOpenedForReading()
+        {
+            // Save test file path so we can refer to the same file throughout the test.
+            string testFilePath = GetTestFilePath();
+
+            // Write random bytes to file so it exists for reading later below.
+            const int FileByteCount = 100;
+            File.WriteAllBytes(testFilePath, RandomNumberGenerator.GetBytes(FileByteCount));
+
+            // Open the file for reading.
+            using (SafeFileHandle handle = File.OpenHandle(testFilePath, FileMode.Open, FileAccess.Read))
+            {
+                // On non-Windows platforms (notably Unix), flushing a file opened for reading should succeed.
+                // It appears that these platforms initially behaved like Windows but then changed their
+                // behavior (see https://www.austingroupbugs.net/view.php?id=671 for the discussion and
+                // https://pubs.opengroup.org/onlinepubs/9699919799/functions/aio_fsync.html for the spec
+                // containing the words "Note that even if the file descriptor is not open for writing...").
+                RandomAccess.FlushToDisk(handle);
```

Comment:
I agree with you @adamsitnik that aligning the behaviours will make life easier for users of this function 👍 

Currently the XML docs I added contain this:

```csharp
        /// <exception cref="T:System.UnauthorizedAccessException">
        /// <paramref name="handle" /> was not opened for writing (on Windows). <-----
        /// </exception>
```

...i.e. they state that if you flush a file opened for reading (for whatever reason), the `UnauthorizedAccessException` is only thrown on Windows. We have three choices here:

1. **Leave it as it is right now:** this means the behaviour is different between platforms but it allows users on Unix to flush files opened for reading. While I can't think of a particular use case for this, it seems to have been important enough to some people to warrant a change in the POSIX spec as per [this discussion](https://www.austingroupbugs.net/view.php?id=671).
2.  **Make Windows behave like Unix:** this means we would catch the `UnauthorizedAccessException` on Windows and ignore it so no error is reported when you flush a file opened for reading on all platforms. This would allow the Unix users to keep the feature and is similar to how we currently ignore errors on Unix when someone flushes a non-seekable file, so we can allow the Windows users to keep that feature in order to flush pipe handles.
3. **Make Unix behave like Windows:** this means we would manually check if a file is opened for reading, then throw `UnauthorizedAccessException` so flushing such a file throws the same error on both platforms. This does give us consistency but it prevents Unix users from doing whatever it is that people had in mind when they [changed the spec](https://www.austingroupbugs.net/view.php?id=671) to allow flushing files opened for reading.

I am still undecided myself but I am leaning towards number 2: **make Windows behave like Unix** which means swallowing errors on Windows so flushing a file opened for reading does NOT throw on either platform. That gives us the consistency we want, while still allowing Unix users to take advantage of that feature (even though we may not be sure what the use case is).

@danmoseley any thoughts here?

Reply:
@adamsitnik thanks for the code snippet! I have made the suggested changed in [this commit](https://github.com/dotnet/runtime/pull/89100/commits/749d9b83d7318521b360fba0a4e2a9e2a8a0fe44) 👍